14 TestComponent/src/components/common/spec/spinner.spec.tsx

```diff
@@ -0,0 +1,63 @@
+import { expect } from 'chai';
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import Spinner from '../spinner';
+
+describe('Spinner presentational component', () =>{
+    it('should renders null and calls to initializeHttp method when passing showSpinner ' +
+        'equals undefined and initializeHttp method like props', () => {
+        let showSpinner = undefined;
+        let initializeHttp = sinon.spy();
+        let spinnerWrapper = shallow(
+            <Spinner showSpinner={showSpinner} initializeHttp={initializeHttp}/>
+        );
+
+        expect(spinnerWrapper.type()).to.be.null;
+        expect(initializeHttp.calledOnce).to.be.true;
+    });
+
+    it('should renders null and calls to initializeHttp method when passing showSpinner ' +
+        'equals null and initializeHttp method like props', () => {
+        let showSpinner = null;
+        let initializeHttp = sinon.spy();
+        let spinnerWrapper = shallow(
+            <Spinner showSpinner={showSpinner} initializeHttp={initializeHttp}/>
+        );
+
+        expect(spinnerWrapper.type()).to.be.null;
+        expect(initializeHttp.calledOnce).to.be.true;
+    });
+
+    it('should renders null and calls to initializeHttp method when passing showSpinner ' +
+        'equals false and initializeHttp method like props', () => {
+        let showSpinner = false;
+        let initializeHttp = sinon.spy();
+        let spinnerWrapper = shallow(
+            <Spinner showSpinner={showSpinner} initializeHttp={initializeHttp}/>
+        );
+
+        expect(spinnerWrapper.type()).to.be.null;
+        expect(initializeHttp.calledOnce).to.be.true;
+    });
+
+    it('should renders spinner and calls to initializeHttp method when passing showSpinner ' +
+        'equals true and initializeHttp method like props', () => {
+        let showSpinner = true;
+        let initializeHttp = sinon.spy();
+        let spinnerWrapper = shallow(
+            <Spinner showSpinner={showSpinner} initializeHttp={initializeHttp}/>
+        );
+
+        expect(spinnerWrapper.contains(
+            <div className="spinnerWrap">
+              <div className="spinnerOverlay"></div>
+              <div className="vertical-offset">
+                <div id="spinner">
+                  Loading...
+                </div>
+              </div>
+            </div>
+        )).to.be.true;
```

Comment:
Let's check that spinnerWrap is present (maybe the content could change in the future but this div will remain)